assets/src/scripts/application-form.js

```diff
@@ -8,3 +8,46 @@ const links = content.getElementsByTagName("a");
     link.rel = "noopener noreferrer";
   }
 });
+
+// Select the textarea element.
+const textAreaElement = document.getElementById("id_study_purpose");
+
+// Select the character counter element.
+const characterCounterElement = document.getElementById("character_counter");
+
+// Select the element that shows the number of characters typed in the textarea.
+const typedCharactersElement = document.getElementById("typed_characters");
+
+// Define the maximum number of characters allowed.
+const maximumCharacters = textAreaElement.getAttribute("max_length");
+
+// Add a "keydown" event listener on the textarea element.
+textAreaElement.addEventListener("keyup", () => {
+  // Count the number of characters typed.
+  const typedCharacters = textAreaElement.value.length;
+
+  /**
+   * Check if the typed characters is more than allowed characters limit.
+   * If yes, then return false.
+   */
+  if (typedCharacters > maximumCharacters) {
+    return false;
```

Comment:
I'm assuming you've tried this out, and if you did something like copy-pasted in a big block of text this would do the right thing (i.e. a keyup event per char in the input, so everything works as intended)?

Reply:
ah, just saw the next commit!

Reply:
Just tested this, and the keyup of paste triggers it 🚀